iib/workers/tasks/build_containerized_add.py

```diff
@@ -0,0 +1,358 @@
+        ``None``.
+    :param dict binary_image_config: the dict of config required to identify the appropriate
+        ``binary_image`` to use.
+    :param list deprecation_list: list of deprecated bundles for the target index image. Defaults
+        to ``None``.
+    :param list build_tags: List of tags which will be applied to intermediate index images.
+    :param str graph_update_mode: Graph update mode that defines how channel graphs are updated
+        in the index.
+    :param dict index_to_gitlab_push_map: the dict mapping index images (keys) to GitLab repos
+        (values) in order to push their catalogs into GitLab.
+    :raises IIBError: if the index image build fails.
+    """
+    reset_docker_config()
+    # Resolve bundles to their digests
+    set_request_state(request_id, 'in_progress', 'Resolving the bundles')
+
+    with set_registry_token(overwrite_from_index_token, from_index, append=True):
+        resolved_bundles = get_resolved_bundles(bundles)
+        verify_labels(resolved_bundles)
+        if check_related_images:
+            inspect_related_images(
+                resolved_bundles,
+                request_id,
+                worker_config.iib_related_image_registry_replacement.get(username),
+            )
+
+    prebuild_info = prepare_request_for_build(
+        request_id,
+        RequestConfigAddRm(
+            _binary_image=binary_image,
+            from_index=from_index,
+            overwrite_from_index_token=overwrite_from_index_token,
+            add_arches=add_arches,
+            bundles=bundles,
+            distribution_scope=distribution_scope,
+            binary_image_config=binary_image_config,
+        ),
+    )
+    from_index_resolved = prebuild_info['from_index_resolved']
+    binary_image_resolved = prebuild_info['binary_image_resolved']
+    arches = prebuild_info['arches']
+    operators = list(prebuild_info['bundle_mapping'].keys())
+    distribution_scope = prebuild_info['distribution_scope']
+
+    index_to_gitlab_push_map = index_to_gitlab_push_map or {}
+    # Variables mr_details, last_commit_sha and original_index_db_digest
+    # needs to be assigned; otherwise cleanup_on_failure() fails when an exception is raised.
+    mr_details: Optional[Dict[str, str]] = None
+    last_commit_sha: Optional[str] = None
+    original_index_db_digest: Optional[str] = None
+
+    Opm.set_opm_version(from_index_resolved)
+
+    _update_index_image_build_state(request_id, prebuild_info)
+    present_bundles: List[BundleImage] = []
+    present_bundles_pull_spec: List[str] = []
+    with tempfile.TemporaryDirectory(prefix=f'iib-{request_id}-') as temp_dir:
+        branch = prebuild_info['ocp_version']
+
+        # Set up and clone Git repository
+        (
+            index_git_repo,
+            local_git_repo_path,
+            localized_git_catalog_path,
+        ) = prepare_git_repository_for_build(
```

Comment:
what happens if `from_index` is not provided? Should it add the bundles to an empty index? Like create-empty-index handles it?